### ISSUE TITLE
linear prediction

### ISSUE THREAD

**Issue by mnezaf** [slideflow/slideflow#388]
Hello,

I plan to train a model for the prediction of a linear label using slideflow. In version 2.3, is it possible to train linear MIL model?

The label I have is in the range of 2.5 to 5.5, which is not balanced, what is the best way in slideflow to split it? I tried to bin the labels and then used this one 
splits = dataset.kfold_split(
    k=3,
    labels='label_bins',
)
The number of labels close to minimum value is much less than those close to mean value and prediction is not correct for the minor label range. I plan to do upsampling for the minor range of labels. Can I use similar augmentation for linear prediction same as here https://slideflow.dev/tutorial7/. 
I would really appreciate if you could provide any documentation for linear prediction.

Thanks in advance

**Comment by jamesdolezal** [comment]
Hello @mnezaf - couple thoughts

## Linear outcomes for MIL
Linear outcome support for MIL is in development but not yet released. We have a working prototype, and depending on how stability testing goes, we may be able to include it in our imminent 2.4.0 release. I will keep you updated.

Regarding data splitting for linear outcomes, the way you've approached it with bins is exactly how I would do it. It would be nice to have a direct option to perform this binning/balancing automatically - I've added it to our backlog. 

## Data augmentation
If you're building a **tile-based model**, you can perform data augmentation (balancing) for a linear outcome based on the bins you've generated. If you're following Tutorial 7, you would replace this line:

```python
train = train.balance('er_status_by_ihc', strategy='category')
```

with:

```python
train = train.balance("label_bins", strategy="category")
```

If you're using the ``Project.train()`` interface, you can specify how balancing is performed using ``balance_headers`` argument:

```python
P.train(..., balance_headers="label_bins")
```

## Data augmentation for MIL
If you're building an **MIL model**, we don't yet have built-in support for data augmentation. This is something on our to-do list (see: #364).

**Comment by jamesdolezal** [comment]
As an update, linear outcomes for MIL models have been implemented and will be available in version 3.0. You can preview this functionality on the `slideflow3` branch.

**Comment by jamesdolezal** [comment]
Support for continuous outcomes has been added in version 3.0
